src/libraries/Common/src/System/CodeDom/CodeTypeReferenceCollection.cs

```diff
@@ -29,7 +29,7 @@ public CodeTypeReferenceCollection(CodeTypeReference[] value)
 
         public CodeTypeReference this[int index]
         {
-            get { return ((CodeTypeReference)(List[index])); }
+            get { return ((CodeTypeReference)(List[index]!)); }
```

Comment:
there is nothing really guarding setter against null

Reply:
consider making most of the overloads nullable

Reply:
The class won't throw an exception, but all the `Add` methods are annotated as taking non-nullable CodeTypeReference instances. So anyone who is using this class shouldn't be putting `null` into it. If they do, that's their bug.

This is the downfall of using the non-generic collection types. If this used `Collection<T>` this wouldn't be a concern.